### ISSUE TITLE
stb_image_write.h: Compiler warning (GCC 10.1)

### ISSUE THREAD

**Issue by mgerhardy** [nothings/stb#1020]
```
../../src/modules/image/stb_image_write.h: In function ‘void stbiw__write1(stbi__write_context*, unsigned char)’:
../../src/modules/image/stb_image_write.h:400:24: warning: comparison of integer expressions of different signedness: ‘int’ and ‘long unsigned int’ [-Wsign-compare]
  400 |    if (s->buf_used + 1 > sizeof(s->buffer))
      |        ~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~
../../src/modules/image/stb_image_write.h: In function ‘void stbiw__write3(stbi__write_context*, unsigned char, unsigned char, unsigned char)’:
../../src/modules/image/stb_image_write.h:408:24: warning: comparison of integer expressions of different signedness: ‘int’ and ‘long unsigned int’ [-Wsign-compare]
  408 |    if (s->buf_used + 3 > sizeof(s->buffer))
      |        ~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~
../../src/modules/image/stb_image_write.h: In function ‘int stbi_write_bmp_to_func(void (*)(void*, void*, int), void*, int, int, int, const void*)’:
../../src/modules/image/stb_image_write.h:502:32: warning: missing initializer for member ‘stbi__write_context::context’ [-Wmissing-field-initializers]
  502 |    stbi__write_context s = { 0 };
      |                                ^
../../src/modules/image/stb_image_write.h:502:32: warning: missing initializer for member ‘stbi__write_context::buffer’ [-Wmissing-field-initializers]
../../src/modules/image/stb_image_write.h:502:32: warning: missing initializer for member ‘stbi__write_context::buf_used’ [-Wmissing-field-initializers]
../../src/modules/image/stb_image_write.h: In function ‘int stbi_write_bmp(const char*, int, int, int, const void*)’:
../../src/modules/image/stb_image_write.h:510:32: warning: missing initializer for member ‘stbi__write_context::context’ [-Wmissing-field-initializers]
  510 |    stbi__write_context s = { 0 };
      |                                ^
../../src/modules/image/stb_image_write.h:510:32: warning: missing initializer for member ‘stbi__write_context::buffer’ [-Wmissing-field-initializers]
../../src/modules/image/stb_image_write.h:510:32: warning: missing initializer for member ‘stbi__write_context::buf_used’ [-Wmissing-field-initializers]
../../src/modules/image/stb_image_write.h: In function ‘int stbi_write_tga_to_func(void (*)(void*, void*, int), void*, int, int, int, const void*)’:
../../src/modules/image/stb_image_write.h:601:32: warning: missing initializer for member ‘stbi__write_context::context’ [-Wmissing-field-initializers]
  601 |    stbi__write_context s = { 0 };
      |                                ^
../../src/modules/image/stb_image_write.h:601:32: warning: missing initializer for member ‘stbi__write_context::buffer’ [-Wmissing-field-initializers]
../../src/modules/image/stb_image_write.h:601:32: warning: missing initializer for member ‘stbi__write_context::buf_used’ [-Wmissing-field-initializers]
../../src/modules/image/stb_image_write.h: In function ‘int stbi_write_tga(const char*, int, int, int, const void*)’:
../../src/modules/image/stb_image_write.h:609:32: warning: missing initializer for member ‘stbi__write_context::context’ [-Wmissing-field-initializers]
  609 |    stbi__write_context s = { 0 };
      |                                ^
../../src/modules/image/stb_image_write.h:609:32: warning: missing initializer for member ‘stbi__write_context::buffer’ [-Wmissing-field-initializers]
../../src/modules/image/stb_image_write.h:609:32: warning: missing initializer for member ‘stbi__write_context::buf_used’ [-Wmissing-field-initializers]
../../src/modules/image/stb_image_write.h: In function ‘int stbi_write_hdr_to_func(void (*)(void*, void*, int), void*, int, int, int, const float*)’:
../../src/modules/image/stb_image_write.h:775:32: warning: missing initializer for member ‘stbi__write_context::context’ [-Wmissing-field-initializers]
  775 |    stbi__write_context s = { 0 };
      |   ../../src/modules/image/stb_image_write.h:775:32: warning: missing initializer for member ‘stbi__write_context::buffer’ [-Wmissing-field-initializers]
../../src/modules/image/stb_image_write.h:775:32: warning: missing initializer for member ‘stbi__write_context::buf_used’ [-Wmissing-field-initializers]
../../src/modules/image/stb_image_write.h: In function ‘int stbi_write_hdr(const char*, int, int, int, const float*)’:
../../src/modules/image/stb_image_write.h:783:32: warning: missing initializer for member ‘stbi__write_context::context’ [-Wmissing-field-initializers]
  783 |    stbi__write_context s = { 0 };
      |                                ^
../../src/modules/image/stb_image_write.h:783:32: warning: missing initializer for member ‘stbi__write_context::buffer’ [-Wmissing-field-initializers]
../../src/modules/image/stb_image_write.h:783:32: warning: missing initializer for member ‘stbi__write_context::buf_used’ [-Wmissing-field-initializers]
../../src/modules/image/stb_image_write.h: In function ‘int stbi_write_jpg_to_func(void (*)(void*, void*, int), void*, int, int, int, const void*, int)’:
../../src/modules/image/stb_image_write.h:1579:32: warning: missing initializer for member ‘stbi__write_context::context’ [-Wmissing-field-initializers]
 1579 |    stbi__write_context s = { 0 };
      |                                ^
../../src/modules/image/stb_image_write.h:1579:32: warning: missing initializer for member ‘stbi__write_context::buffer’ [-Wmissing-field-initializers]
../../src/modules/image/stb_image_write.h:1579:32: warning: missing initializer for member ‘stbi__write_context::buf_used’ [-Wmissing-field-initializers]
../../src/modules/image/stb_image_write.h: In function ‘int stbi_write_jpg(const char*, int, int, int, const void*, int)’:
../../src/modules/image/stb_image_write.h:1588:32: warning: missing initializer for member ‘stbi__write_context::context’ [-Wmissing-field-initializers]
 1588 |    stbi__write_context s = { 0 };
      |                                ^
../../src/modules/image/stb_image_write.h:1588:32: warning: missing initializer for member ‘stbi__write_context::buffer’ [-Wmissing-field-initializers]
../../src/modules/image/stb_image_write.h:1588:32: warning: missing initializer for member ‘stbi__write_context::buf_used’ [-Wmissing-field-initializers]
                             ^

```

**Comment by RufUsul** [comment]
stb_image_write.h: fixed some more warnings (GCC v8.3.0) https://github.com/nothings/stb/pull/1029

**Comment by rygorous** [comment]
The sign compare warnings are fixed in the dev branch. The missing field initializers I don't get on my WSL Ubuntu 20.04 (gcc 9.3.0), and they are actually somewhat hard to fix without introducing other problems. Either way, this issue was reported many times; closing as it's a duplicate.
